net/HttpRequest.hpp

```diff
@@ -1757,6 +1778,25 @@ class Session final : public ProtocolHandlerInterface
         net::asyncConnect(_host, _port, isSecure(), shared_from_this(), pushConnectCompleteToPoll);
     }
 
+    void asyncConnect(const std::weak_ptr<SocketPoll>& poll)
+    {
+        _socket.reset(); // Reset to make sure we are disconnected.
```

Comment:
I worry about Sockets that are not owned by SocketPolls unless they are very briefly in transit between them - or we are taking a shared_ptr rather briefly for some purpose.